### PR TITLE
TaskManager now inherits from ComponentManager. 

### PR DESCRIPTION
**Pull request by bjwhite-fnal**
Also added SourceManager, ChannelManager, and SourceSubscriptionManager files for future integration.

### CODE REVIEW

**Comment by jcpunk** [comment]
For the various formatting complaints, I'd just run it through `black`

File: src/decisionengine/framework/managers/ComponentManager.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import importlib
+import logging
+import multiprocessing
+import uuid
+
+from decisionengine.framework.dataspace import dataspace
+from decisionengine.framework.dataspace import datablock
+from decisionengine.framework.taskmanager.ProcessingState import State, ProcessingState
+
+def create_runner(module_name, class_name, parameters):
```
Comment (jcpunk) [inline src/decisionengine/framework/managers/ComponentManager.py:15]: It doesn't appear that this is covered by unit tests.

File: src/decisionengine/framework/managers/ComponentManager.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        :type data: :obj:`dict`
+        :arg data: key, value pairs
+        :type header: :obj:`~datablock.Header`
+        :arg header: data header
+        :type data_block: :obj:`~datablock.DataBlock`
+        :arg data_block: data block
+        """
+
+        if not isinstance(data, dict):
```
Comment (jcpunk) [inline src/decisionengine/framework/managers/ComponentManager.py:82]: It doesn't appear that this is covered by unit tests.

**Review by knoepfel — COMMENTED** [review]
Brandon can you rebase this PR off of the current HEAD of `master`?  Also there are still some `structlog` changes that will need to be adopted  in the `ChannelManager`, but overall looks okay.

File: src/decisionengine/framework/managers/ChannelManager.py [hunk 3]
```diff
@@ -0,0 +1,434 @@
+"""
+Channel Manager
+"""
+import threading
+import logging
+import structlog
+import time
+import queue
+
+import pandas
```
Comment (jcpunk) [inline src/decisionengine/framework/managers/ChannelManager.py:10]: pandas-vet recommends `import pandas as pd`

**Review by knoepfel — APPROVED** [review]
Thanks, Brandon.  With the move to `celery`, it's not clear what adjustments will need to be made down the road, but your contributions here have greatly helped us understand the complexities of the source-proxy replacement.  The component manager concept is a great help for us.

**Pull request merged**